src/r_bsp.c

```diff
@@ -383,6 +383,11 @@ static void R_AddLine (seg_t *line)
   //      should already be interpolated.
   R_MaybeInterpolateSector(backsector);
 
+  if (backsector->heightsec != -1)
```

Comment:
Hm, could you maybe integrate this into `R_FakeFlat()`? It contains the same check for `backsector->heightsec != -1`. Just an idea. 😉

Reply:
> Is this compatible with the fixes to all the other issues that have been posted before?

It seems so, I copied this from https://github.com/fabiangreffrath/woof/commit/6fda34f497da756cad2f50dbefecc7e719fef03a

> Hm, could you maybe integrate this into R_FakeFlat()?

Not sure how to integrate it into `R_FakeFlat()`, but we can simplify a bit:
```diff
diff --git a/src/r_bsp.c b/src/r_bsp.c
index 99c4ecf..f9fe955 100644
--- a/src/r_bsp.c
+++ b/src/r_bsp.c
@@ -189,13 +189,9 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
       // Replace floor and ceiling height with other sector's heights.
       tempsec->floorheight   = s->floorheight;
       tempsec->ceilingheight = s->ceilingheight;
-      tempsec->interpfloorheight   = s->interpfloorheight;
-      tempsec->interpceilingheight = s->interpceilingheight;
 
       // killough 11/98: prevent sudden light changes from non-water sectors:
       if (underwater && (tempsec->  floorheight = sec->floorheight,
-			 tempsec->interpfloorheight = sec->interpfloorheight,
-			 tempsec->interpceilingheight = s->interpfloorheight-1,
 			 tempsec->ceilingheight = s->floorheight-1, !back))
         {                   // head-below-floor hack
           tempsec->floorpic    = s->floorpic;
@@ -207,7 +203,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
             if (s->ceilingpic == skyflatnum)
               {
                 tempsec->floorheight   = tempsec->ceilingheight+1;
-                tempsec->interpfloorheight = tempsec->interpceilingheight+1;
                 tempsec->ceilingpic    = tempsec->floorpic;
                 tempsec->ceiling_xoffs = tempsec->floor_xoffs;
                 tempsec->ceiling_yoffs = tempsec->floor_yoffs;
@@ -236,8 +231,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
           {   // Above-ceiling hack
             tempsec->ceilingheight = s->ceilingheight;
             tempsec->floorheight   = s->ceilingheight + 1;
-            tempsec->interpceilingheight = s->interpceilingheight;
-            tempsec->interpfloorheight   = s->interpceilingheight + 1;
 
             tempsec->floorpic    = tempsec->ceilingpic    = s->ceilingpic;
             tempsec->floor_xoffs = tempsec->ceiling_xoffs = s->ceiling_xoffs;
@@ -246,7 +239,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
             if (s->floorpic != skyflatnum)
               {
                 tempsec->ceilingheight = sec->ceilingheight;
-                tempsec->interpceilingheight = sec->interpceilingheight;
                 tempsec->floorpic      = s->floorpic;
                 tempsec->floor_xoffs   = s->floor_xoffs;
                 tempsec->floor_yoffs   = s->floor_yoffs;
@@ -262,6 +254,8 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
               *ceilinglightlevel = s->ceilinglightsec == -1 ? s->lightlevel :
               sectors[s->ceilinglightsec].lightlevel; // killough 4/11/98
           }
+      tempsec->interpfloorheight   = s->interpfloorheight;
+      tempsec->interpceilingheight = s->interpceilingheight;
       sec = tempsec;               // Use other sector
     }
   return sec;
```

Reply:
Okay, then it's good. 😁